test/unit/active_record_view_model/has_many_test.rb

```diff
@@ -695,7 +711,7 @@ def test_functional_update_update_failure
       ParentView.deserialize_from_view(update_view)
     end
 
-    assert_match(/Stale update/, ex.message)
+    assert_match(/Stale functional update/, ex.message)
```

Comment:
This exception needs to be accurately communicated to the client. I'd expect a test for that here. See also other comment.